src/nhs_neighbor_lists.jl

```diff
@@ -0,0 +1,115 @@
+@doc raw"""
+    NeighborListsNeighborhoodSearch{NDIMS}(search_radius, n_particles;
```

Comment:
Maybe **PrecomputedNeighborhodSearch** would make it easier to understand in the actual application of this method?

Reply:
Yes, this is good! `NeighborListNeighborhoodSearch` is way too cumbersome. @LasNikas ?

Reply:
Yes, that is better